### ISSUE TITLE
[API] Improve response of stock transaction endpoints

### ISSUE THREAD

**Issue by MrJohnZoidberg** [grocy/grocy#769]
**Description:**
This occurs if the product has multiple stock entries. The amount in the API response is in this case the amount which was substracted from *only one* of the stock entries.

**Examples:**
In the English demo version the product "cold cuts" has 4 stock entries, each with 1 as amount.
I consume 4 cold cuts, but the response only contains the amount -1. Expected: -4
The same bug applies to the open action.

Another one with tare weight handling enabled: Flour has the amount 2000g in stock, with two stock entries, each with 1000g. When I consume 2000g (in the field I write 500g because of tare weight), the response conains 1000g as substracted amount. Expected: 2000g

**Off-topic:**
(I need the correct amount in the response for calculating the new amount of the product without making a stock request again.)

A huge "thank you" for continously developing this great software!

**Comment by berrnd** [comment]
I don't consider this a bug, it's maybe odd, but behaved never differently. Also thought there is already an issue for this, but haven't found it. I haven't changed that so far because that would be a breaking change in the API, changes at least all purchase/consume/inventory/transfer/open endpoints. And maybe the whole stock handling needs to be changed for that.

**TLDR:**
Purchase/consume/inventory/transfer/open endpoints return an `stock_log` object, but the only relevant and maybe useful response field is `transaction_id`.

Longer explanation:
When adding a product to stock, it get's added to the `stock` table "as is". Means when you add an amount of 5, there is one row with that amount. If you add 5 times 1, there are 5 rows with an amount of 1 each, but they are tied together by a `stock_id`. It's both the same total value, row count in the `stock` table doesn't matter.

When removing something, there is a loop through all `stock` rows ordered by the rule "First expiring first, then first in first out" (unless something specific, by `stock_id`, is selected). So by that loop `stock` rows get maybe splitted or multiple rows get removed. For each `stock_id` an `stock_log` row is created (currently not really needed, like a lot of "log"-things, but #64 is hopefully ready before I die and for that it'll be most probably helpful), but they all get the same `transaction_id`.

And the API response is then just the _last_ created `stock_log` entry.

Maybe a simple solution is to only respond that `transaction_id`. Or all `stock_log` rows for that `transaction_id`, so that the client would be able to summarize the total amount. But both would be a breaking change.

> (I need the correct amount in the response for calculating the new amount of the product without making a stock request again.)

- You can not rely on that nothing else was removed/added/opened/etc. in the meantime
- If the request is successful, the requested amount was removed/added/opened/etc.
- => Just query the new stock amount after posting the stock transaction, that's also how the frontend currently does that...

**Comment by MrJohnZoidberg** [comment]
Thank you really much for the long explanation, I will use your proposal :)

**Comment by berrnd** [comment]
Because the API is already kind of completely broken for the next release (on Tuesday), I now also changed this - excerpt from the changelog:

>   - The following endpoints now return all bookings of the transaction (so the response is now an array, was before a single stock booking - and a random one if the transaction affected multiple stock entries)
>     - PUT `/stock/entry/{entryId}`
>     - POST `/stock/products/{productId}/add`
>     - POST `/stock/products/{productId}/consume`
>     - POST `/stock/products/{productId}/transfer`
>     - POST `/stock/products/{productId}/inventory`
>     - POST `/stock/products/{productId}/open`
>     - POST `/stock/products/by-barcode/{barcode}/add`
>     - POST `/stock/products/by-barcode/{barcode}/consume`
>     - POST `/stock/products/by-barcode/{barcode}/transfer`
>     - POST `/stock/products/by-barcode/{barcode}/inventory`
>     - POST `/stock/products/by-barcode/{barcode}/open`
>     - (The response is the same as if you would fetch the stock transaction via `/stock/transactions/{transactionId}`)
